### ISSUE TITLE
psych engine limit question

### ISSUE THREAD

**Issue by TerrorGaming473** [ShadowMario/FNF-PsychEngine#5966]
### What is your question?

what is the music limit of Psych Engine? I was wanting to make a really big song

**Comment by fox-64** [comment]
is 25 minutes and 21 seconds i think 
source: https://youtu.be/zBM2ql3myoE?t=519

**Comment by Spongeyboi** [comment]
The reason why is due to file size. The game caches the song into memory, and a file that big will seriously impact performance or freeze your pc. This was as high as I can get it before my pc froze, which it did after 12 seconds.
![image](https://user-images.githubusercontent.com/71215126/156033408-9e934f1c-eebd-4324-9a6e-28526b53341b.png)

**Comment by SoreKing** [comment]
Alright I got intrigued by this question and I did my own testing.

The maximum possible song length I could get in a single song is 23 min 18 sec or more precisely

1398101 milliseconds.
![Screenshot 2022-03-02 203230](https://user-images.githubusercontent.com/71531145/156497012-ae84d494-9619-4f60-952b-92358cc8bf6b.png)

Anymore than that will overflow the song length counter in game into the negatives. Once that happens, the song and chart are unplayable as the beat and the step turn negative and break the chart. The game does still run and I can exit out.
![Screenshot 2022-03-02 203351](https://user-images.githubusercontent.com/71531145/156497218-4cb32369-64b3-4b5a-a1f9-2091403b35b6.png)

Also I found that I could load a 1 hr 30 min ogg track in the game without crashing but the counter is wrong and its still unplayable. I have a pretty beefy pc though so your mileage may vary. 

This was done on the latest version of psych engine on github as of March 2, 2022.
